src/FlagProvider.tsx

```diff
@@ -1,67 +1,62 @@
+/** @format */
+
 import * as React from 'react';
 import FlagContext from './FlagContext';
 import { UnleashClient, IConfig, IContext } from 'unleash-proxy-client';
 
+type eventArgs = [Function, any];
+
 interface IFlagProvider {
-  config: IConfig;
+  config?: IConfig;
+  unleashClient?: UnleashClient;
 }
 
-const FlagProvider: React.FC<IFlagProvider> = ({ config, children }) => {
-  const [client, setClient] = React.useState(null);
-  const functionCalls = React.useRef<any>();
-  functionCalls.current = [];
-
-  React.useEffect(() => {
-    const client = new UnleashClient(config);
-
-    functionCalls.current.forEach((call: any) => {
-      call(client);
-    }, []);
+const FlagProvider: React.FC<IFlagProvider> = ({
+  config,
+  children,
+  unleashClient,
+}) => {
+  const client = React.useRef<UnleashClient>(unleashClient);
 
-    functionCalls.current = [];
+  if (!config && !unleashClient) {
+    console.warn(
+      `You must provide either a config or an unleash client to the flag provider. If you are initializing the client in useEffect, you can avoid this warning by
+      checking if the client exists before rendering.`
+    );
+  }
 
-    setClient(client);
+  if (!client.current) {
+    client.current = new UnleashClient(config);
+  }
 
-    client.start();
+  React.useEffect(() => {
+    client.current.start();
   }, []);
 
   const updateContext = async (context: IContext): Promise<void> => {
-    if (!client) {
-      deferCall(async (client: any) => await client.updateContext(context));
-      return;
-    }
-    await client.updateContext(context);
-  };
-
-  const deferCall = (callback: (client: any) => void) => {
-    functionCalls.current.push(callback);
+    await client.current.updateContext(context);
   };
 
   const isEnabled = (name: string) => {
-    if (!client) {
-      deferCall((client: any) => client.isEnabled(name));
-      return;
-    }
-    return client.isEnabled(name);
+    return client.current.isEnabled(name);
   };
 
   const getVariant = (name: string) => {
-    if (!client) {
-      deferCall((client: any) => client.getVariant(name));
-      return {};
-    }
-    return client.getVariant(name);
+    return client.current.getVariant(name);
+  };
+
+  const on = (event: string, ...args: eventArgs) => {
+    return client.current.on(event, ...args);
```

Comment:
I know this was there from before, but could you explain what this actually achieve?

Reply:
...Maybe a way to defer event registrations?

Reply:
Are you referring to the deferCall functionality? Or the on function?

Reply:
@ivarconr Ignoring the defer functionality since that's gone now, I understand this just as proxy functions, exposing the functionality of the client without exposing the actual client. So the downstream component is able to pull a function out of context rather than the client, which has a function...

It's a small thing; just a design preference IMO.